frontend/src/modules/entitydetails/selectors.js

```diff
@@ -0,0 +1,50 @@
+/* @flow */
+
+import { createSelector } from 'reselect';
+
+import * as navigation from 'core/navigation';
+import * as plural from 'core/plural';
+import * as entitieslist from 'modules/entitieslist';
+
+import type { Navigation } from 'core/navigation';
+import type { Entities } from 'modules/entitieslist';
+
+
+const entitiesSelector = (state): string => state[entitieslist.NAME].entities;
+
+
+export function _getTranslationForSelectedEntity(
+    entities: Entities,
+    params: Navigation,
+    pluralForm: number,
+): string {
+    const entityId = params.entity;
+    const entity = entities.find(element => element.pk === entityId);
+    if (pluralForm === -1) {
+        pluralForm = 0;
+    }
+    if (entity && entity.translation[pluralForm].string && !entity.translation[pluralForm].rejected) {
+        return entity.translation[pluralForm].string;
+    }
+    return '';
+}
+
+
+/**
+ * Return the active translation string for the currently selected entity
+ * and plural form.
+ *
+ * The active translation is either the approved one, the fuzzy one, or the
+ * most recent non-rejected one.
```

Comment:
That's true, but the algorithm for that is on backend, so I wonder if it makes sense to keep the comment here.

Well, it probably doesn't hurt, so if you read the frontend code, you don't need to trace it all the way to python. :)